packages/dd-trace/src/noop/span.js

```diff
@@ -9,15 +9,8 @@ class NoopSpan extends Span {
     super()
 
     // Avoid circular dependency
-    Object.defineProperties(this, {
```

Comment:
should we have a comment here on the optimization? why not to use `Object.defineProperties`, so someone doesn't come through and add it back in the future?

Reply:
A direct assignment will actually cause the circular dependency that the `Object.defineProperties` was trying to avoid when logging. We'd need to make other changes for this to work safely.

Reply:
I think storing the `noop` reference on the trace instead and then just a boolean or a type check on the `NoopSpanContext` would fix the circular dependency since noop spans do not have any reference to the trace.

Reply:
After discussing this offline, it should be fine as long as we don't use `JSON.stringify` on the span.

Reply:
@brettlangdon It's not specifically *here* that `Object.defineProperties` is expensive, but basically anywhere that's in a hot path. At this point, I'd almost suggest making a lint rule for this.